airbyte-api/src/main/openapi/config.yaml

```diff
@@ -768,6 +768,69 @@ paths:
           description: Connection not found
         "422":
           $ref: "#/components/responses/InvalidInput"
+  /v1/scheduler/sources/check_connection:
+    post:
+      tags:
+        - scheduler
+      summary: Run check connection for a given source configuration
+      operationId: executeSourceCheckConnection
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SourceCreate"
```

Comment:
Mentioning this since we eventually want to support API-native access to Airbyte not just through the UI...

strictly speaking we we don't need the `name` field from `SourceCreate`. We could create a `SourceConnectionParams` that doesn't contain that. We could even have `SourceCreate` extend it like described here https://json-schema.org/understanding-json-schema/structuring.html#id5

IDK if it's worth the trouble for this PR but it makes this API endpoint only ask for the information it requires

Reply:
it's a good point. going to see how long it will take me to to do it.